apps/web/app/api/slack/callback/route.ts

```diff
@@ -0,0 +1,67 @@
+import { getServerSession } from "next-auth";
+import { NextRequest, NextResponse } from "next/server";
+
+import { authOptions } from "@formbricks/lib/authOptions";
+import { WEBAPP_URL } from "@formbricks/lib/constants";
+import { createOrUpdateIntegration } from "@formbricks/lib/integration/service";
+import { TIntegrationSlackConfig, TIntegrationSlackCredential } from "@formbricks/types/integration/slack";
+
+export async function GET(req: NextRequest) {
+  const url = req.url;
+  const queryParams = new URLSearchParams(url.split("?")[1]); // Split the URL and get the query parameters
+  const environmentId = queryParams.get("environment"); // Get the value of the 'state' parameter
+
+  const session = await getServerSession(authOptions);
+
+  if (!session || !environmentId) {
+    console.log("there is either no session or environementid");
+    return {
+      redirect: {
+        destination: "/",
+        permanent: false,
+      },
+    };
+  }
+
+  // @ts-expect-error
+  const { slack } = session;
+
+  if (!slack) {
+    console.log("slack session is not defined");
+    return {
+      redirect: {
+        destination: "/",
+        permanent: false,
+      },
+    };
+  }
+
```

Comment:
Do you think we can remove the logs here?
Also, there can be an `error` queryParam in case of any error in integrating Slack, can we also handle that?